ovn_k8s/processor/policy_processor.py

```diff
@@ -28,34 +30,192 @@ def __init__(self, pool):
+        if ns_events[0] == ns_events[-1]:
+            # Single event
+            ns_data = ns_events[-1].metadata
+            isolated_final = ns_data.get('custom', {}).get('isolated', False)
+        else:
+            ns_data_final = ns_events[-1].metadata
+            isolated_final = ns_data_final.get(
+                'custom', {}).get('isolated', False)
+            ns_data_initial = ns_events[0].metadata
+            isolated_initial = ns_data_initial.get(
+                'custom', {}).get('isolated', False)
+            if isolated_final == isolated_initial:
+                # Nothing actually changed
+                log.dbg("Pod: %s (Namespace: %s): initial and final "
+                        "namespace isolation status are identical (%s) - "
+                        "no processing necessary" %
+                        (pod_name. namespace, isolated_final))
+                return
+        self.mode.remove_acls(pod_id)
+        if not isolated_final:
+            log.dbg("Pod: %s (Namespace: %s): not isolated, "
+                    "whitelisting traffic for pod" % (pod_name, namespace))
+            self.mode.whitelist_pod_traffic(pod_id, namespace, pod_name)
+
+    def _apply_pod_acls(self, pod_data, pod_events, policies):
+        # Do shallow copy of pod events as the list will be modified
+        pod_events = pod_events[:]
+        pod_name = pod_data['metadata']['name']
+        pod_ns = pod_data['metadata']['namespace']
+
+        log.dbg("Pod: %s (Namespace: %s): applying ACLs..." % (
+            pod_name, pod_ns))
+        # Check for a namespace event. This might mean that there has
+        # been a translation from isolated to non-isolated and therefore
+        # all we have to do would be whitelisting traffic
+        self._apply_pod_ns_acls(pod_data, pod_events)
+        if kubernetes.is_namespace_isolated(
+                variables.K8S_API_SERVER, pod_ns):
+            # If we are here the namespace is isolated, and policies must
+            # be translated into acls
+            self.mode.apply_pod_policy_acls(pod_data, policies)
+        log.dbg("Pod: %s (Namespace: %s): ACLs applied" % (pod_name, pod_ns))
 
     def process_events(self, events):
         log.dbg("Processing %d events from queue" % len(events))
-        for event in events[:]:
+        affected_pods = {}
+        pod_events = {}
+        for event in events:
             if isinstance(event, processor.NSEvent):
                 # namespace add -> create policy watcher
                 # namespace delete -> destory policy watcher
                 # namespace update -> check isolation property
-                self._process_ns_event(event)
-            events.remove(event)
+                self._process_ns_event(event, affected_pods, pod_events)
+            if isinstance(event, processor.NPEvent):
+                # policy add -> create ACLs for affected pods
+                # policy delete -> remove ACLs for affected pods
```

Comment:
The function _process_np_event does not actually create and destroy ACLs for the pods right? All it does is marks all the affected pods. And then removes/creates address sets and removes/creates pseudo ACLs